src/main/java/com/glencoesoftware/bioformats2raw/Converter.java

```diff
@@ -1224,8 +1232,20 @@ public void saveResolutions(int series)
                 futures.add(future);
                 executor.execute(() -> {
                   try {
-                    processChunk(series, resolution, plane, xx, yy, zz,
-                                 width, height, depth);
+                    int[] shape = {1, 1, 1, height, width};
+                    int[] offset;
+                    IFormatReader reader = readers.take();
+                    try {
+                      String o = new StringBuilder(
+                          dimensionOrder != null? dimensionOrder.toString()
+                          : reader.getDimensionOrder()).reverse().toString();
+                      shape[o.indexOf("Z")] = depth;
```

Comment:
I'm not sure I understand why ```o.indexOf("Z")``` is used here when line 1025 above assumes ```shape[2]``` is the Z size.